src/error.rs

```diff
@@ -4,14 +4,24 @@
 /// Any error generated by Ardite or it‘s drivers should be output using this
 /// type. This allows for a comprehensive display of the error when a service
 /// reports it to the user.
-pub struct Error {
+pub struct Error<'a> {
   /// A specific error code which describes the error.
   code: ErrorCode,
   /// A message providing more detail beyond the error code.
-  message: String,
+  message: &'a str,
   /// A hint to the user on what to do next to try and avoid the error
   /// happening again. This is optional.
-  hint: Option<String>
+  hint: Option<&'a str>
+}
+
+impl<'a> Error<'a> {
+  pub fn unimplemented(message: &'a str) -> Self {
```

Comment:
We can add another function, I just wanted a convenience function for writing unimplemented errors.